FormidableLabs/nuka-carousel
# chore: build and use built types in package.json

### Description

Currently, the index.d.ts file is a separate set of types that must be manually maintained. This is more work than needed, as we already have types in `types.ts`. This change adds a job to build the types as part of the build process, and then changes package.json to use those types.

#### Type of Change

Please delete options that are not relevant.


### How Has This Been Tested?

1. Run `yarn build-v5`
2. Install the package locally in my own project using `yarn add ../nuka-carousel`
3. Verify that the types are present

### Checklist: (Feel free to delete this section upon completion)

- [X] My code follows the style guidelines of this project (I have run `yarn lint`)
- [ ] New and existing unit tests pass locally with my changes (I have run `yarn test` and `yarn test-e2e`)
  - The existing tests seem broken, even before this change
